modules/hix-project.nix

```diff
@@ -40,8 +40,10 @@
     };
   };
 
-  # Default values for other project options (things that do not have defaults for non hix projects)
-  config = {
-    compiler-nix-name = lib.mkDefault "ghc8107";
+  # Default value for compiler-nix-name (does not have a default for non hix projects).
+  # Stack projects do not require a default as the `resolver` in the `stack.yaml`
+  # specifies one.
+  config = lib.mkIf (!config ? "stackYaml") {
+    compiler-nix-name = lib.mkDefault "ghc945";
```

Comment:
That would be nicer, but `compiler-nix-name` is in the `project` level module.  I think there is a `compiler.nix-name` in the other `modules`, but we don't currently use it to set the `project` level `compiler-nix-name` (only the other way around).

Is it ok if we merge this for now and improve it if we can in the refactoring of the `modules`?

Reply:
> Is it ok if we merge this for now and improve it if we can in the refactoring of the modules?

Sure, go head. Sorry for making you wait, I wrote that comment just as you opened the PR but I didn't realise I had "started a review" rather than "add a single comment". I'll approve now.